zookeeper-server/src/main/java/org/apache/zookeeper/server/SnapshotComparer.java

```diff
@@ -0,0 +1,455 @@
+    SnapshotComparer app = new SnapshotComparer();
+    app.compareSnapshots(args);
+  }
+
+  private void compareSnapshots(String[] args) throws Exception {
+    CommandLine parsedOptions;
+    try {
+      parsedOptions = new BasicParser().parse(options, args);
+    } catch (ParseException e) {
+      usage();
+      return;
+    }
+    File left = (File)parsedOptions.getParsedOptionValue(leftOption);
+    File right = (File)parsedOptions.getParsedOptionValue(rightOption);
+    int byteThreshold = Integer.parseInt((String)parsedOptions.getParsedOptionValue(byteThresholdOption));
+    int nodeThreshold = Integer.parseInt((String)parsedOptions.getParsedOptionValue(nodeThresholdOption));
+    boolean debug = parsedOptions.hasOption(debugOption);
+    boolean interactive = parsedOptions.hasOption(interactiveOption);
+    System.out.println("Successfully parsed options!");
+    TreeInfo leftTree = new TreeInfo(left);
+    TreeInfo rightTree = new TreeInfo(right);
+
+    System.out.println(leftTree.toString());
+    System.out.println(rightTree.toString());
+
+    compareTrees(leftTree, rightTree, byteThreshold, nodeThreshold, debug, interactive);
+  }
+
+  private static class TreeInfo {
+    public static class TreeNode {
+      final String label;
+      final long size;
+      final List<TreeNode> children;
+      long descendantSize;
+      long descendantCount;
+
+      private TreeNode() {
+        label = null;
+        size = -1;
+        children = null;
+        descendantSize = -1;
+        descendantCount = -1;
+      }
+
+      public static class AlphabeticComparator implements Comparator<TreeNode> {
+        public int compare(TreeNode left, TreeNode right) {
+          if (left == right) {
+            return 0;
+          }
+          if (left == null) {
+            return -1;
+          }
+          if (right == null) {
+            return 1;
+          }
+          return left.label.compareTo(right.label);
+        }
+      }
+
+      public TreeNode(String label, long size) {
+        this.label = label;
+        this.size = size;
+        this.children = new ArrayList<TreeNode>();
+      }
+
+      void AddChild(TreeNode child) {
+        children.add(child);
+      }
+
+      void populateChildren(String path, DataTree dataTree, TreeInfo treeInfo) throws Exception {
+        populateChildren(path, dataTree, treeInfo, 1);
+      }
+
+      void populateChildren(String path, DataTree dataTree, TreeInfo treeInfo, int currentDepth) throws Exception {
+        List<String> childLabels = null;
+        childLabels = dataTree.getChildren(path, null, null);
+
+        if (childLabels != null && !childLabels.isEmpty()) {
+          for (String childName : childLabels){
+            String childPath = path + "/" + childName;
+            DataNode childNode = dataTree.getNode(childPath);
+            long size = childNode.data == null ? 0 : childNode.data.length;
+            TreeNode childTreeNode = new TreeNode(childPath, size);
```

Comment:
quick comment: 
Finbugs complians about this and the  `ICAST_IDIV_CAST_TO_DOUBLE`, `VA_FORMAT_STRING_USES_NEWLINE`

Reply:
Good catch, there are a few find bug warnings here need address. Will update PR with the fix.